Fix for "Invalid relative media source path shows root path contents"

JoshuaLuckers opened the pull request:
### What does it do?
An invalid relative media source path would result in '/' being set as path.
With this PR there is a check if realpath returns an actual path instead of false.

### Why is it needed?
Quick solution for issue #14024 

### Related issue(s)/PR(s)
#14024 

This also needs to be fixed in 3.x

JoshuaLuckers committed:
If realpath returns false don't set the pathAbsolute to '/'

JoshuaLuckers commented:
This is a quick solution. There are various improvements that can be made.
I also found some other bugs. I will fill new issues for those at a later moment.

Mark-H commented:
What is `$bases['pathAbsolute']` set to if `$realpath === false`?

Mark-H commented:
The failure on travis was due to travis, not the change. Rebuild marked is as green now.

JoshuaLuckers commented:
> What is `$bases['pathAbsolute']` set to if `$realpath === false`?

I believe it's set to nothing (iirc elsewhere it checks if `$bases['pathAbsolute']` is empty. If yes it returns an empty array.

Jako committed:
Improvements

- Use dot concatenation instead of curly braces. Same usage as in line 69. 
- Set $bases['pathAbsolute'] to empty string if the realpath is wrong.

Jako commented:
Checked. Works fine.

OptimusCrime commented:
Does this PR also close #14024? @Jako

Jako commented:
Yes it did.
